### PR TITLE
Auth integration

### PR DESCRIPTION
**Pull request by MiguelAngelGutierrezMaya**
# Auth Integration PR Description

### Core changes

- Introduced the `@features/auth` module with domain models, use cases, Cognito repository integration, and a centralized `AuthProvider` to manage session restore, refresh, sign-in, sign-out, sign-up, account confirmation, password recovery, forced password updates, and MFA/TOTP flows.
- Added the missing auth routes in Expo Router for register confirmation, forgot-password confirmation, new-password challenge, MFA verification, MFA setup, and OAuth callback handling.
- Added route guards so authenticated users are redirected away from auth screens and unauthenticated users are redirected away from dashboard routes.
- Reworked the registration and recovery flows to support confirmation codes, resend actions, password strength validation, optional phone capture, terms/privacy consent, and localized error handling.
- Added the Cognito OAuth/PKCE foundation required for social sign-in. The infrastructure and repository support are in place, while the in-app callback completion is still intentionally limited until the `codeVerifier` persistence is finalized.

### Schema changes

- No database schema changes.
- Updated Cognito infrastructure to export the hosted UI domain so it can be consumed by other stacks.
- Updated Amplify hosting to inject `COGNITO_DOMAIN` and `COGNITO_REGION` into the app build environment.
- Updated Expo app configuration to support environment-specific app identifiers, secure storage/web browser plugins, and EAS build/submit profiles for development and production.

### Minor changes

- Added reusable auth and UI components such as OTP input, password strength feedback, QR code rendering, phone input, info popup, avatar, and shared text input primitives.
- Expanded English and Spanish translations for authentication and dashboard-related copy.
- Updated environment loading so `client/main` and `packages/transactional` receive the expected env files per environment.
- Added automated coverage for the new auth domain, infrastructure, provider, routes, templates, and app configuration.

## Screenshots/Recordings

Screenshot 2026-03-15 at 01 02 10
Screenshot 2026-03-15 at 01 02 18

## How to Test

1. Load the environment for the target app build and confirm the auth variables are available to `client/main`, especially `EXPO_PUBLIC_USER_POOL_ID`, `EXPO_PUBLIC_USER_POOL_CLIENT_ID`, `EXPO_PUBLIC_COGNITO_DOMAIN`, `EXPO_PUBLIC_COGNITO_REGION`, and `EXPO_PUBLIC_APPLICATION_URL`.
2. Start the app with `pnpm --filter @client/main start` for native or `pnpm --filter @client/main web` for web.
3. Register a new account with email and password. Optionally enable phone number and accept the terms/privacy consent.
Expected result: the app redirects to the confirmation screen and allows resending the verification code with cooldown feedback.
4. Complete account confirmation with the code received by email or SMS.
Expected result: a success state is shown and the user is redirected back to the login screen.
5. Sign in with the verified account.
Expected result: the user lands on the dashboard, and revisiting auth routes redirects back to the dashboard while the session is active.
6. Sign out, or try to access a dashboard route without an active session.
Expected result: protected dashboard routes redirect to `/auth/login`.
7. Run the forgot-password flow for an existing account and complete the reset with a valid code and a new valid password.
Expected result: the user is redirected back to login and can authenticate with the updated password.
8. Test with a Cognito user in `FORCE_CHANGE_PASSWORD`.
Expected result: the app redirects to the new-password screen and completes the challenge before granting access.
9. Test with a Cognito user that requires MFA or TOTP setup.
Expected result: the app redirects to the MFA verification or MFA setup screen and grants access after successful completion.
10. For web validation, confirm the deployed build receives `COGNITO_DOMAIN` and `COGNITO_REGION` and boots without missing-auth-configuration errors.

Notes:
- Social sign-in buttons and Cognito OAuth plumbing are included in this branch, but end-to-end callback completion is not part of the current manual validation scope yet.

## Checklist

- [x] The PR description includes clear instructions on how to test the implementation manually.

##### Added Tests?

- [x] yes
  - [x] Verified that this PR is not creating duplicate use cases
  - [ ] All edge cases described in the acceptance criteria are covered, and tests have clear and descriptive names
- [ ] no, because they aren't needed
- [ ] no, because I need help

_Override these when needed._

```sh
pnpm --filter @features/auth exec jest --runInBand
pnpm --filter @client/main exec jest --runInBand
```

### CODE REVIEW

File: .envrc [hunk 1]
```diff
@@ -22,19 +22,34 @@ echo "🚀 Loading environment variables from .env.local"
...
 
 if [ "$ENV" = "production" ]; then
     echo "🚀 Loading environment variables from .env.production"
     [[ -f ./config/.env.production ]] && dotenv ./config/.env.production
+
+    echo "🚀 Copying environment variables to packages/transactional"
     [[ -f ./config/.env.production ]] && cp ./config/.env.production ./packages/transactional/.env
+
+    echo "🚀 Loading environment variables on client/main"
+    [[ -f ./client/main/.env.production ]] && cp ./client/main/.env.production ./client/main/.env.development.local
```
[the remark this answers is not in the document]
Comment (MiguelAngelGutierrezMaya) [reply .envrc:52]: At the moment, we need to persist this code as it is because the tests are local to this filename
